src/make-gecko-action-hooks.js

```diff
@@ -66,33 +66,22 @@ module.exports.run = async function(options) {
 };
 
 const makeHookDetails = (taskclusterYml, action) => {
-  const repoUrlExpression = '${payload.decision.repository.url[8:]}';
-  const actionOverrides = {
-    repo_scope: {
-      $let: {
-        repoUrl: {
-          $if: 'payload.decision.repository.url[-1] == "/"',
-          then: {$eval: 'payload.decision.repository.url[:-1]'},
-          else: {$eval: 'payload.decision.repository.url'},
-        },
-      },
-      in: 'assume:repo:${repoUrl[8:]}:action:' + action.actionPerm,
-    },
-  };
-
-  if (action.actionPerm !== 'generic') {
-    // enforce that action.cb_name == actionPerm
-    actionOverrides.cb_name = action.actionPerm;
-  }
-
   const task = {
     $let: {
       tasks_for: 'action',
       action: {
-        $merge: [
-          {$eval: 'payload.decision.action'},
-          actionOverrides,
-        ],
+        name: '${payload.decision.action.name}',
+        title: '${payload.decision.action.title}',
+        description: '${payload.decision.action.description}',
+        taskGroupId: '${payload.decision.action.taskGroupId}',
+        // repo_scope is calculated in the hook's JSON-e rendering; note that this is limited
+        // by the hook's `hook-id:..` role, too
+        repo_scope: 'assume:repo:${payload.decision.repository.url[8:]}:action:' + action.actionPerm,
```

Comment:
It took me a few minute to convince myself that this was safe. I think it would make sense to specifically call out that this can only allow scopes that are also given to the `hook-id:*` role, so that the validation of the repository url happens there.

Reply:
Thanks - updated

Reply:
:+1: